opencti-platform/opencti-front/src/__tests__/components/Security.test.tsx

```diff
@@ -0,0 +1,251 @@
+import AppIntlProvider from '../../components/AppIntlProvider';
+import { BYPASS, EXPLORE_EXUPDATE, KNOWLEDGE_KNUPDATE } from '../../utils/hooks/useGranted';
+import { UserContext } from '../../utils/hooks/useAuth';
+
+const userAdmin = {
+  name: 'admin',
+  user_email: 'admin@opencti.io',
+  firstname: 'Admin',
+  lastname: 'OpenCTI',
+  language: 'auto',
+  unit_system: 'auto',
+  theme: 'default',
+  external: true,
+  userSubscriptions: {
+    edges: [],
+  },
+  capabilities: [{ name: BYPASS }],
+};
+
+const userZeroCapability = {
+  name: 'admin',
+  user_email: 'admin@opencti.io',
+  firstname: 'Admin',
+  lastname: 'OpenCTI',
+  language: 'auto',
+  unit_system: 'auto',
+  theme: 'default',
+  external: true,
+  userSubscriptions: {
+    edges: [],
+  },
+  capabilities: [],
+};
+
+const userOneCapability = {
+  name: 'admin',
+  user_email: 'admin@opencti.io',
+  firstname: 'Admin',
+  lastname: 'OpenCTI',
+  language: 'auto',
+  unit_system: 'auto',
+  theme: 'default',
+  external: true,
+  userSubscriptions: {
+    edges: [],
+  },
+  capabilities: [{ name: KNOWLEDGE_KNUPDATE }],
+};
+
+const userTwoCapability = {
+  name: 'admin',
+  user_email: 'admin@opencti.io',
+  firstname: 'Admin',
+  lastname: 'OpenCTI',
+  language: 'auto',
+  unit_system: 'auto',
+  theme: 'default',
+  external: true,
+  userSubscriptions: {
+    edges: [],
+  },
+  capabilities: [{ name: KNOWLEDGE_KNUPDATE }, { name: EXPLORE_EXUPDATE }],
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const AdminContext: any = { me: userAdmin, settings: {}, bannerSettings: {}, entitySettings: {}, platformModuleHelpers: {}, schema: {} };
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const ZeroCapabilityContext: any = { me: userZeroCapability, settings: {}, bannerSettings: {}, entitySettings: {}, platformModuleHelpers: {}, schema: {} };
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const OneCapabilityContext: any = { me: userOneCapability, settings: {}, bannerSettings: {}, entitySettings: {}, platformModuleHelpers: {}, schema: {} };
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const TwoCapabilityContext: any = { me: userTwoCapability, settings: {}, bannerSettings: {}, entitySettings: {}, platformModuleHelpers: {}, schema: {} };
+
+let container: HTMLDivElement | null;
+describe('Security validations', () => {
+  it('admin with BYPASS should be allowed whatever the required permission is.', async () => {
+    act(() => {
+      const createComponent = (divElement: HTMLDivElement, environment: RelayMockEnvironment) => {
+        ReactDOM.createRoot(divElement).render(
+          <RelayEnvironmentProvider environment={environment}>
+            <AppIntlProvider settings={{ platform_language: 'auto' }}>
+              <ThemeProvider theme={createTheme()}>
+                <UserContext.Provider value={AdminContext}>
+                  <Security needs={[KNOWLEDGE_KNUPDATE, EXPLORE_EXUPDATE]}>
```

Comment:
Well done for the tests ! It could be nice to have a helper test component like `<TestApp...>` that instantiate all those providers to avoid duplicating those lines that pollute the tests readings